packages/dds/tree/src/test/simple-tree/object.spec.ts

```diff
@@ -41,160 +46,194 @@ export function testObjectPrototype(proxy: object, prototype: object) {
-					)) {
-						it(`Object.getOwnPropertyDescriptor(${pretty(
-							initialTree,
-						)}, ${key}) -> ${pretty(descriptor)}`, () => {
-							const root = hydrate(schema, initialTree);
-							assert.deepEqual(
-								Object.getOwnPropertyDescriptor(findObjectPrototype(root), key),
-								descriptor,
-								`Proxy must expose Object.prototype.${key}`,
-							);
-						});
-					}
+				it(`${pretty(initialTree)}.isPrototypeOf(root) -> false`, () => {
+					const root = hydrate(schema, initialTree);
+					const asObject = root as object;
+					// eslint-disable-next-line no-prototype-builtins -- compatibility test
+					assert.equal(asObject.isPrototypeOf(asObject), false);
 				});
+			});
 
-				describe("methods inherited from Object.prototype", () => {
-					it(`${pretty(initialTree)}.isPrototypeOf(Object.create(root)) -> true`, () => {
-						const root = hydrate(schema, initialTree);
-						const asObject = root as object;
-						// eslint-disable-next-line no-prototype-builtins -- compatibility test
-						assert.equal(asObject.isPrototypeOf(Object.create(asObject)), true);
-					});
+			describe(`${pretty(initialTree)}.propertyIsEnumerable`, () => {
+				for (const key of Object.getOwnPropertyNames(initialTree)) {
+					const expected = Object.prototype.propertyIsEnumerable.call(initialTree, key);
 
-					it(`${pretty(initialTree)}.isPrototypeOf(root) -> false`, () => {
+					it(`${key} -> ${expected}`, () => {
 						const root = hydrate(schema, initialTree);
 						const asObject = root as object;
 						// eslint-disable-next-line no-prototype-builtins -- compatibility test
-						assert.equal(asObject.isPrototypeOf(asObject), false);
+						assert.equal(asObject.propertyIsEnumerable(key), expected);
 					});
-				});
+				}
+			});
+		}
+	});
 
-				describe(`${pretty(initialTree)}.propertyIsEnumerable`, () => {
-					for (const key of Object.getOwnPropertyNames(initialTree)) {
-						const expected = Object.prototype.propertyIsEnumerable.call(
-							initialTree,
-							key,
-						);
+	function test1(fn: (subject: object) => unknown) {
+		for (const { schema, initialTree } of testCases) {
+			const real = structuredClone(initialTree) as object;
+			const expected = fn(real);
 
-						it(`${key} -> ${expected}`, () => {
-							const root = hydrate(schema, initialTree);
-							const asObject = root as object;
-							// eslint-disable-next-line no-prototype-builtins -- compatibility test
-							assert.equal(asObject.propertyIsEnumerable(key), expected);
-						});
-					}
-				});
-			}
+			it(`${pretty(real)} -> ${pretty(expected)}`, () => {
+				const proxy = hydrate(schema, initialTree);
+				const actual = fn(proxy as object);
+				assert.deepEqual(actual, expected);
+			});
+		}
+	}
+
+	describe("keys", () => {
+		describe("Object.keys", () => {
+			test1((subject) => Object.keys(subject));
 		});
 
-		function test1(fn: (subject: object) => unknown) {
+		describe("for-in", () => {
```

Comment:
This diff got a little crazy because I moved tests across describe blocks. The only meaningful changes are the new test suites here and just below: "for-in" and "key in"

Reply:
You  should test Reflect.ownKeys: I'm pretty sure it doesn't handle empty fields correctly. Object.getOwnPropertyNames will also fail I think, but Reflect.ownKeys is the core thing the others are based on with proxies and what shouldbe tested

Reply:
I added a test for ownKeys.